src/components/TxData.js

```diff
@@ -198,12 +213,67 @@ class TxData extends React.Component {
     if (uid === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid) {
       return hathorLib.constants.HATHOR_TOKEN_CONFIG.symbol;
     }
-    const tokenConfig = this.state.tokens.find((token) => token.uid === uid);
+    const tokenConfig = this.props.transaction.tokens.find((token) => token.uid === uid);
     if (tokenConfig === undefined) return '';
     return tokenConfig.symbol;
   }
 
+  /**
+   * Returns if the token from uid in parameter is not registered in the wallet
+   *
+   * @param {string} uid UID of the token to check
+   *
+   * @return {boolean} If token is unknown (not registered)
+   */
+  isTokenUnknown = (uid) => {
+    const tokenConfig = this.state.tokens.find((token) => token.uid === uid);
+    if (tokenConfig === undefined) return false;
```

Comment:
When can this happen? Shouldn't `this.state.tokens` have all tokens from this screen?

Reply:
Yes. I was just doing extra validation in case something really weird happens. Because if by any chance `tokenConfig` it would raise js exception and the wallet would stop.

I feel like it's extra care but doesn't hurt having it. What do you think?

Reply:
No problem.